pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -347,9 +347,16 @@ func buildLocalityLbEndpoints(env *model.Environment, proxyNetworkView map[strin
 	localityLbEndpoints := make([]endpoint.LocalityLbEndpoints, 0, len(lbEndpoints))
 
 	for locality, eps := range lbEndpoints {
+		var weight uint32
+		for _, ep := range eps {
+			weight += ep.LoadBalancingWeight.GetValue()
+		}
 		localityLbEndpoints = append(localityLbEndpoints, endpoint.LocalityLbEndpoints{
 			Locality:    util.ConvertLocality(locality),
 			LbEndpoints: eps,
+			LoadBalancingWeight: &types.UInt32Value{
+				Value: weight,
```

Comment:
If you delete this code:
* Unit tests still pass
* Output (manually tested) remains identical

So it doesn't seem like this does anything.

The reason is that we have `weight := float64(originalWeight*weight) / float64(totalWeight)` but originalWeight is always equal to totalWeight.

The tests are always good to have, but it doesn't seem like we need this change.

If we do need to change the behavior, then can you add some tests that capture the change in behavior?

Reply:
That's because I set the same weight for `region1/zone1/subzone1` and `region1/zone1/subzone2`

Reply:
I pulled down your change she tested manually, it didn't seem to have any impact. I was testing with different weights. If this actually does change the behavior can you change the tests to show that?

Reply:
I will update the test

Reply:
Now you can test again, to see if you remove the above code

Reply:
:+1: ok got it now, thanks!

My concern is that we have two ways to determine the traffic split - what we currently do, and this change. I personally don't think either is wrong, they just behave differently and we need to explicitly decide how they should behave.

I wrote up a quick explanation of the two options and the impact they have, hopefully it is clear. Please take a look: https://gist.github.com/howardjohn/0116ed5df015c5a9a7aece378055f93b

Lets get a consensus on what the behavior should be